arduino-ide-extension/src/browser/serial/monitor/serial-monitor-send-input.tsx

```diff
@@ -6,6 +6,75 @@ import { BoardsServiceProvider } from '../../boards/boards-service-provider';
 import { MonitorModel } from '../../monitor-model';
 import { Unknown } from '../../../common/nls';
 
+class RingList {
+  protected ring: string[];
+  protected size: number;
+  protected begin: number;
+  protected index: number;
+  protected end: number;
+
+  constructor(size: number = 100) {
+    this.Init = this.Init.bind(this);
+    this.Push = this.Push.bind(this);
+    this.Prev = this.Prev.bind(this);
+    this.Next = this.Next.bind(this);
+    this.Init(size);
+  }
+
+  public Init(size: number = 100)
+  {
+    this.ring = [];
+    this.size = (size > 0) ? size : 1;
+    this.begin = 0;
+    this.index = 0;
+    this.end = -1;
+  }
+
+  public Push(val: string): number {
```

Comment:
Method names start with lowercase and have camel case styling. Please change to `init`, `prev`, etc.

Reply:
The `public` keywords are redundant. Please remove them. It should be:

```ts
push(val: string): number {
  // code
}

prev(): string {
  // code
}
```

Reply:
Done